### PR TITLE
Enable GCS based bazel cache

### PR DESCRIPTION
**Pull request by anakinxc**

### CODE REVIEW

**Review by tongke6 — DISMISSED** [review]
LGTM

File: .circleci/regtest-config.yml [hunk 1]
```diff
@@ -33,36 +33,18 @@ jobs:
...
-          command: sh devtools/decompress-build-cache.sh engine_regtest_cache
       - run:
           name: "Build binary"
           command: |
             set +e 
             mkdir -p image
             cp docker/scql.Dockerfile image
             docker run -d -it --name scql-dev -v $(pwd):/home/admin/dev/ -v ~/.cache/engine_regtest_cache:/root/.cache/engine_regtest_cache -w /home/admin/dev --cap-add=SYS_PTRACE --security-opt seccomp=unconfined --cap-add=NET_ADMIN --privileged=true secretflow/release-ci:latest
-            docker exec -it scql-dev bash -c "make && bazel build //engine/... -c opt --ui_event_filters=-info,-debug,-warning --disk_cache=~/.cache/engine_regtest_cache" 
+            docker exec -it scql-dev bash -c "make && bazel build //engine/... -c opt --ui_event_filters=-info,-debug,-warning"
```
Comment (jingshi-ant) [inline .circleci/regtest-config.yml:43]: 1.why the gcs not work for regtest?
2. dir in line42: ~/.cache/engine_regtest_cache can be cleaned.

Comment (anakinxc) [reply .circleci/regtest-config.yml:43]: > 1.why the gcs not work for regtest? 2. dir in line42: ~/.cache/engine_regtest_cache can be cleaned.

It works, just we do not provide cache for build based on release-ci at this point.

**Review by jingshi-ant — APPROVED** [review]
LGTM

**Review by tongke6 — APPROVED** [review]

**Pull request merged**